presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -204,6 +213,30 @@ public ResourceGroupInfo getInfo()
         }
     }
 
+    public ResourceGroupInfo getStaticResourceGroupInfo()
+    {
+        synchronized (root) {
+            return new ResourceGroupInfo(
```

Comment:
Should we add a static resource group check invariant here. If somehow we make a call to this function and it is not static, then it should return a null (preferably an `Optional.empty()`) or raise an `UnsupportedOperationException` ?

Reply:
Makes sense. I am going to add it.